cstar/tests/test_roms_marbl_example.py

```diff
@@ -0,0 +1,102 @@
+import builtins
+from pathlib import Path
+import yaml
+
+import cstar
+import pytest
+
+import cstar
+##
+from pathlib import Path
+import shutil
+from contextlib import contextmanager
+
+@pytest.fixture
+def mock_user_input():
+    """Monkeypatch wich will automatically respond "y" to any call for input"""
+    @contextmanager
+    def _mock_input(input_string):
+        original_input = builtins.input
+        def mock_input_function(_):
+            return input_string
+        builtins.input = mock_input_function
+        try:
+            yield
+        finally:
+            builtins.input = original_input
+    
+    return _mock_input
+
+
+# TODO we want to make this unnecessary, ideally by defining fixtures that create temporary directories for all of these
+## Delete output of any previous run of this script
+for oldfiles in ["local_input_files/",
+                 "local_additional_code/",
+                 "roms_marbl_example_case/",
+                 "roms_marbl_local_case/",
+                 "test_blueprint.yaml",
+                 "modified_test_blueprint.yaml",
+                 "test_blueprint_local.yaml"]:
+    
+    oldpath=Path(oldfiles)
+    if oldpath.exists():
+        if oldpath.is_dir():
+            shutil.rmtree(oldpath)
+        else:
+            oldpath.unlink()
```

Comment:
We also want cstar to be robust to these files existing / not existing, which means we want to test with and without automatic deletion.

@dafyddstephenson are these directories all configurable at runtime in cstar? Because if so I could make a `tmpdir` for each of them and pass that in.

Reply:
> @dafyddstephenson are these directories all configurable at runtime in cstar? 
Not sure I understand what you mean by this, let me know if this helps: The directory names are set by the calls to `Case.from_blueprint` later in the script using the `caseroot` argument. C-Star creates the directory if it doesn't already exist. You could create a temporary directory and use it as long as you supply it as a `Path` to `Case.from_blueprint()` using the `caseroot` argument.

Reply:
> You could create a temporary directory and use it as long as you supply it as a Path to Case.from_blueprint() using the caseroot argument.

Okay great, that's exactly what I wanted to know (and what I'm already doing in this PR!)

Reply:
I think this cleaning up works now 🍾

Reply:
To clarify, right now if I run pytest on a freshly-cloned cstar then the only things that are still around after the test are `cstar_local_config.py` and `cstar/externals`. I think the latter is unavoidable, but the former can be removed via refactoring hopefully (or at least leave only a machine-specific configuration `.yml` file).

The test passes, and passes even if I run it again with no other actions.